### ISSUE TITLE
Card header/footer doesn't look nice with primary + inverse

### ISSUE THREAD

**Issue by roest01** [twbs/bootstrap#18946]
Combine `card-primary` and `card-inverse` like written in [documentation](http://v4-alpha.getbootstrap.com/components/card/#inverted-text) results in an unfinished view.

![bootstrap-card-header](https://cloud.githubusercontent.com/assets/2870718/12447802/42e0d322-bf72-11e5-816c-4e85955e30cb.png)

https://jsfiddle.net/roest/LL0qnnxm/

---

Also think about making the header or footer primary results in problem with card-inverse selector wich is compiled to `.card-inverse .card-header` therefore not a good result with `card-header card-primary card-inverse` 

![bootstrap-header-primary](https://cloud.githubusercontent.com/assets/2870718/12448049/c7482c86-bf73-11e5-8161-8b9d54aa4612.png)

https://jsfiddle.net/roest/ndLrxy54/

**Comment by twbs-lmvtfy** [comment]
Hi @roest01!

You appear to have posted a live example (https://fiddle.jshell.net/roest/ndLrxy54/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 60, column 9: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @roest01!

You appear to have posted a live example (https://fiddle.jshell.net/roest/LL0qnnxm/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 64, column 9: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
> therefore not a good result with card-header card-primary card-inverse

I'm reasonably sure that it's only permissible to use `.card-inverse`/`.card-primary` on `.card` itself, not on subcomponents like `.card-header`, so your 2nd example seems invalid.

**Comment by roest01** [comment]
thank you for you'r answer cvrebert

my second example aims to the [contextual alternatives](https://getbootstrap.com/components/#panels) of bootstrap v3 panel `panel panel-primary`

![bootstrapv3_panel](https://cloud.githubusercontent.com/assets/2870718/12449311/9cd65654-bf7c-11e5-8f5e-637ade5bd2ab.png)

by definition you're right - `.card-primary` should be on `.card` itself - but then how can we separate between filled primary card and header primary card?

---

In my view the design of old `panel-primary` is the more flexible / simpler element because you doesn't have to take care about the [content + the font color design](https://cloud.githubusercontent.com/assets/2870718/12449739/a3882790-bf7f-11e5-86cb-7d429cb411d5.png).

**Comment by cvrebert** [comment]
As mentioned in [the bleeding edge version of the migration guide](https://github.com/twbs/bootstrap/blob/v4-dev/docs/migration.md#panels), you can simply add `.bg-primary` to `.card-header` to achieve basically the same look as the old `.panel-primary`.

**Comment by roest01** [comment]
confirmed - [works as expected](https://jsfiddle.net/roest/ndLrxy54/3/). Therefore my second example is invalid.

With this information i've also been able to create a [better looking first example](https://jsfiddle.net/roest/LL0qnnxm/3/). It's not perfect (different border color, multiple "primary" declaration) but you can work with it. 

It's your decision to close this ticket if this usage is best practice and no more need to change the code

Thank you :)

**Comment by rwaltenberg** [comment]
It solves by adding a background-color: transparent to card-header/card-footer when .card-inverse'd (see 	f23ba61)

**Updated fiddle:** https://jsfiddle.net/LL0qnnxm/16/

```css
.card-inverse .card-header, 
.card-inverse .card-footer { 
  background-color: transparent; 
  border-color: rgba(255, 255, 255, 0.2); 
} 
```

**Comment by mdo** [comment]
Merged #21111 so closing.
